src/session/session_api.c

```diff
@@ -1114,6 +1114,73 @@ __session_drop_readonly(
 err:	API_END_RET(session, ret);
 }
 
+/*
+ * __session_import --
+ *	WT_SESSION->import method.
+ */
+static int
+__session_import(WT_SESSION *wt_session,
+    const char *uri, const char *path, const char *config)
+{
+	WT_DECL_RET;
+	WT_SESSION_IMPL *session;
+	char *value;
+
+	WT_UNUSED(config);
+
+	session = (WT_SESSION_IMPL *)wt_session;
+	SESSION_API_CALL_NOCONF(session, import);
+
+	WT_ERR(__wt_inmem_unsupported_op(session, NULL));
+
+	if (!WT_PREFIX_MATCH(uri, "file:") &&
+	    !WT_PREFIX_MATCH(uri, "table:"))
+		WT_ERR(__wt_bad_object_type(session, uri));
+	if (!__wt_absolute_path(path))
+		WT_ERR_MSG(session, EINVAL,
+		    "WT_SESSION.import requires %s be an absolute path",
+		    path);
+
+	ret = __wt_metadata_search(session, uri, &value);
+	if (ret == 0)
+		WT_ERR_MSG(session, EINVAL,
+		    "an object named \"%s\" already exists in the database",
+		    uri);
+	WT_ERR_NOTFOUND_OK(ret);
```

Comment:
We don't do anything to prevent importing a file that's already in the database, which would mean you could possible have a single file appear multiple times, by different names, in the database metadata. Nothing good is going to happen after that, but I couldn't think of any easy way to detect it.